index.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * @module mapeo-schema
+ */
+
+import { Observation } from './types/proto/observation.js'
+import assert from 'node:assert'
+import fs from 'node:fs'
+import b4a from 'b4a'
+import Ajv from 'ajv'
+import addFormats from 'ajv-formats'
+
+const loadJSON = (path) =>
+  JSON.parse(fs.readFileSync(new URL(path, import.meta.url)))
+const ajv = new Ajv()
+addFormats(ajv)
+
+const schemaTypesMap = {
+  observation: {
+    blockPrefix: 'obse', // 4 byte string
+    protobufSchema: Observation,
+    jsonSchema: loadJSON('./schema/observation.json'),
+  },
+}
+/**
+ * Encode a an object validated against a schema as a binary protobuf to send to an hypercore.
+ * @param {import('./types/schema/observation')} obj - Object to be encoded
+ * @returns {Buffer} protobuf encoded buffer with 2 bytes prepended, one for the type of record and the other for the version of the schema */
+export const encode = (obj) => {
+  const record = Object.assign({}, obj)
+  const recordType = schemaTypesMap[record.type]
+  const validate = ajv.compile(recordType.jsonSchema)
+  const isValid = validate(record)
+  assert(isValid, JSON.stringify(validate.errors, true, 2))
```

Comment:
```suggestion
  assert(isValid, JSON.stringify(validate.errors, null, 2))
```
putting `true` as the second arg is invalid.